### DISCUSSION TITLE
Round trip Reservation - Return leg check-in past the first leg trip date

### DISCUSSION THREAD

**Discussion by karjalaj** [jdholtz/auto-southwest-check-in#260] [Q&A] [answered]
The below script ran on past the first leg trip, but the script is still trying to check-in for prior completed trips (Date/Time condition check may require after retrieving the reservation details)
$ python southwest.py
Successfully scheduled the following flights to check in for ABC XYZ:
Flight from St. Louis to Houston (Hobby) on 2024-05-03 3:25 PM CDT
Flight from Houston (Hobby) to St. Louis on 2024-05-10 9:30 PM CDT

Checking in to flight from 'St. Louis' to 'Houston (Hobby)' for ABC XYZ

Successfully scheduled the following flights to check in for AAA BBB:
Flight from Houston (Hobby) to St. Louis on 2024-05-11 5:15 AM CDT

Failed to check in to flight CCCCCC for ABC XYZ. Reason: Too Many Requests 429.
Check in at this url: https://mobile.southwest.com/check-in

Successfully scheduled the following flights to check in for BBB CCC:
Flight from St. Louis to Houston (Hobby) on 2024-05-03 3:25 PM CDT
Flight from Houston (Hobby) to St. Louis on 2024-05-10 9:30 PM CDT

Checking in to flight from 'St. Louis' to 'Houston (Hobby)' for BBB CCC

Failed to check in to flight CCCCCC for BBB CCC. Reason: Bad Request 400.
Check in at this url: https://mobile.southwest.com/check-in

**Comment by karjalaj** [comment, marked as answer]
Looks like there is a departed flight branch currently in development stage and hopefully the merge to master may fix this issue.

**Reply by jdholtz** [reply]
To apply the fix until it’s merged, you can run `git pull && git checkout develop && pip install -r requirements.txt` and the rerun the script to make sure it checks you in correctly. I’ll merge the fix soon in a new minor update.

**Comment by karjalaj** [comment]
Thank you very much. Quick question - Today I checked-in a flight to las
vegas, but got C10. It tried many times as per log but failed. Also it
started trying from 5 seconds before the actual time and then kept making
many attempts. Any way to change that setting? or how to find the root
cause?

**Reply by babehboi** [reply]
So usually it will start 5 sec before and keep attempting just to cushion and account for time differences. 

I'm honestly not sure what you'd want to change it to as normally it should work. What errors were you getting was it not getting a get response or did you get a different error? We just got back from a trip that was Friday and Saturday return

**Reply by jdholtz** [reply]
The 429 you are receiving is probably what’s seen in #201 but with people have been getting the same error with check-ins more often now. 

> Any way to change that setting? or how to find the root cause?

The script is configured to start checking in 5 seconds early in case the Southwest servers are a little ahead. It’s also to hopefully make the check-in go through as early as possible. There’s no way to configure that setting though.

**Comment by karjalaj** [comment]
Here is the log, not sure where it went wrong as the final completion was
26 secs later which delayed the process to get C10

2024-05-12 13:35:22 DEBUG Process-1:1[checkin_handler:105]: Lock released
2024-05-12 13:35:22 DEBUG Process-1:1[checkin_handler:109]: Sleeping until
check-in: 1772 seconds...
2024-05-12 14:04:55 DEBUG Process-1:1[checkin_handler:128]: Attempting to
check in
2024-05-12 14:04:55 DEBUG Process-1:1[checkin_handler:156]: Submitting
check-in with a POST request
2024-05-12 14:04:55 DEBUG Process-1:1[checkin_handler:188]: Making GET
request to check in
2024-05-12 14:05:15 DEBUG Process-1:1[utils:34]: Successfully made request
after 5 attempts
2024-05-12 14:05:15 DEBUG Process-1:1[checkin_handler:194]: Making POST
request to check in
2024-05-12 14:05:26 DEBUG Process-1:1[utils:34]: Successfully made request
after 5 attempts
2024-05-12 14:05:26 DEBUG Process-1:1[checkin_handler:164]: Successfully
checked in after 1 attempts
2024-05-12 14:05:26 DEBUG Process-1:1[notification_handler:103]: Sending
successful check-in notification...

**Reply by jdholtz** [reply]
As you received a 429 earlier (in your original post), you were probably getting a 429 for the first 4 attempts for each part of the check-in too. There’s no known workaround or fix for this yet.

**Comment by karjalaj** [comment]
The original post is for trip from Houston to STL 2 days before. This one C10 is from STL to Las Vegas (ran the script today without any 429 response), but the log shows 6 attempts.

**Reply by jdholtz** [reply]
It didn’t fail with a 429 because it tries 20 times and successfully checked in on the 6th attempt. It’s unknown for sure (as it isn’t logged) but you probably hit a few 429 errors as that’s unfortunately been a pretty common issue people have been running into lately.

**Comment by karjalaj** [comment]
I did GIT PULL and GIT CHECKOUT for latest updates from DEVELOP branch, but the CHECKIN didn't go through for one of the flights leaving LAS VEGAS to STL tomorrow 5:05 PM PDT. See the below logs, Looks like it did tried 20 attempts, but got the error as "Something went wrong" (I used -debug--screenshots option)

The time difference of Attempting to check-in and Submitting Check-in with POST request is almost 3 minutes.

2024-05-15 16:36:18 DEBUG Process-1:1[checkin_handler:105]: Lock released
2024-05-15 16:36:19 DEBUG Process-1:1[checkin_handler:109]: Sleeping until check-in: 1715 seconds...
2024-05-15 17:04:54 DEBUG Process-1:1[checkin_handler:128]: Attempting to check in
2024-05-15 17:07:46 DEBUG Process-1:1[checkin_handler:156]: Submitting check-in with a POST request
2024-05-15 17:07:46 DEBUG Process-1:1[checkin_handler:188]: Making GET request to check in
2024-05-15 17:08:19 DEBUG Process-1:1[utils:58]: Failed to make request after 20 attempts: Too Many Requests 429
2024-05-15 17:08:19 DEBUG Process-1:1[utils:60]: Response body: {
    "code": 429999999,

    "message": "Error.",

    "messageKey": "ERROR",

    "httpStatusCode": "BAD_REQUEST",

    "requestId": "",

    "infoList": []
   }
2024-05-15 17:08:19 DEBUG Process-1:1[checkin_handler:137]: Failed to check in. Error: Too Many Requests 429. Exiting
2024-05-15 17:08:19 DEBUG Process-1:1[notification_handler:112]: Sending failed check-in notification...
2024-05-15 17:09:03 DEBUG Process-1[checkin_handler:58]: Stopping check-in for current flight
2024-05-15 17:09:03 DEBUG Process-1[checkin_handler:61]: Killing process with PID 14668

**Reply by babehboi** [reply]
It's unfortunately the nature of the beast. Errors 429 and 403 are happening lately. Are you using reservation numbers or logging into your account for these check-ins?

**Reply by jdholtz** [reply]
The log message `Submitting check-in with a POST request` is actually an artifact which shouldn't be there (I just fixed it locally). As for why it started the GET request 3 minutes after printing `Attempting to check in`, I am not sure. None of the lines of code in between those two messages should cause any delay (just a `print` call and moving into a function where the second log message is printed). Perhaps it is due to your computer not running that process for a long time. In that case, I don't know if this script has much control over that.

Did you get a statement outputted to the terminal saying something like `Checking in to flight from 'XX' to 'YY' for First Last` after the log message `Attempting to check in`? If you had notifications set up, did you also get the notification at 17:08?

**Reply by karjalaj** [reply]
I used the CONFIG file as input for entering the confirmation number with first and last name.

PS C:\Users\kjpra\GIT\auto-southwest-check-in> python southwest.py --debug-screenshots

Successfully scheduled the following flights to check in for ABC YYY:
Flight from Las Vegas to St. Louis on 2024-05-16 3:05 PM PDT

Checking in to flight from 'Las Vegas' to 'St. Louis' for ABC YYY

Failed to check in to flight CCCCCC for ABC YYY. Reason: Too Many Requests 429.
Check in at this url: https://mobile.southwest.com/check-in

Stopping monitoring for reservation with confirmation number CCCCCCC and name ABC YYY
Cancelling check-in from 'Las Vegas' to 'St. Louis' for ABC YYY

**Reply by jdholtz** [reply]
Did you have notifications set up in the config file too? If you did, did you also receive the notification at the delayed time (17:08) instead of the actual check-in time (at about 17:05)?

**Reply by karjalaj** [reply]
No, I don't have notification setup in my config file. Just the below lines (modified the confirmation number and name)
{
    "check_fares": false,
    "notification_urls": "",
    "notification_level": 1,
    "retrieval_interval": 24,
    "accounts": [],
    "reservations": [
        {"confirmationNumber": "CCCCCC", "firstName": "ABC", "lastName": "YYY"}
    ]
}

**Reply by jdholtz** [reply]
If you run into this problem again (to see it is reproducible) you can file an issue and we can look more into it at that point.
